browser/app/js/objects/actions.js

```diff
@@ -222,19 +222,37 @@ export const deleteCheckedObjects = () => {
 
 export const shareObject = (object, days, hours, minutes) => {
   return function (dispatch, getState) {
+    const hasServerDomain = hasServerPublicDomain(getState())
     const currentBucket = getCurrentBucket(getState())
     const currentPrefix = getCurrentPrefix(getState())
     const objectName = `${currentPrefix}${object}`
     const expiry = days * 24 * 60 * 60 + hours * 60 * 60 + minutes * 60
     if (web.LoggedIn()) {
       return web
-        .PresignedGet({
-          host: location.host,
-          bucket: currentBucket,
-          object: objectName,
-          expiry: expiry,
+        .GetBucketPolicy({ bucketName: currentBucket, prefix: currentPrefix })
```

Comment:
For a bucket without any policy, `web.GetBucketPolicy` is throwing the following error

```
No bucket policy found for bucket: test1
```

Due to this error, an object cannot be shared.

Reply:
Fixed it by adding catch block which returns "fake" policy object